apollo-compiler/src/test/graphql/com/example/case_sensitive_enum/kotlin/responseBased/case_sensitive_enum/TestQuery.kt.expected

```diff
@@ -37,7 +37,7 @@ public class TestQuery() : Query<TestQuery.Data> {
   public override fun selections(): List<CompiledSelection> = TestQuerySelections.root
 
   public data class Data(
-    public val enum_: Enum?
+    public val `enum`: Enum?
```

Comment:
I must admit I don't understand this one. Is this KotlinPoet automatically escaping it? If yes why is it not done in other places?

Reply:
Relevant PR from kotlinpoet: https://github.com/square/kotlinpoet/pull/994

Feels weird escaping in both Apollo + KotlinPoet but I'm not sure we can rely 100% on KotlinPoet because we might use identifiers in places where KotlinPoet doesn't know enough context to escape? 

Things like:

```kotlin
CodeBlock.of("%L.value", layout.fieldName(responseName))
```

Might be worth trying to remove all escaping from Apollo and give KotlinPoet the proper hints?

Reply:
Done in bc1aa75283549ead8f067e4bee39b0543beeeaf7 and e3aeb455729db79ed8df904ace4097a25dc5bf0a.

I added an integration test with keywords in different situations, and it looks like it's working. I hope some other cases of `%T` needing to be `%N` aren't missing!